rfcs/0005-managed-attributes-image-automation/README.md

```diff
@@ -0,0 +1,82 @@
+# RFC-0005 Managed attributes on Image Automation
+
+**Status:** provisional
+
+**Creation date:** 2021-12-16
+
+**Last update:** 2021-12-16
+
+## Summary
+
+Image automation controller can update some attributes of a kubernetes object. 
+Today this is limited to image name, image tag and image name+tag. 
+This RFC aims to extend this functionality to other attributes.  
+
+## Motivation
+
+Some automation or observability tools can use label to identify better a 
+kubernetes object. It can be linked to a version, to a date, to a code 
+origin... For multiple reason, the image tag can reflect poorly this
+data. An example can be given by the image reflector controller which 
+can extract a part of the tag and use it to sort and select the correct one.
+
+### Goals
+
+This RFC aims to describe a way to extract such additional value from the
+image tag, and to use them to update some attributes on the kubernetes object.
+
+### Non-Goals
+
+This RFC will focus on image automation controller. It is a non goal to extend
+this to manually modified kubernetes objects.
+
+## Proposal
+
+### User Stories
+
+As a user, I can update the filter pattern on the image policy object to 
+capture additional data. 
+Then, I can reference the name of the captured group in the comment of a 
+kubernetes object so that the attribute linked to this comment can be updated. 
+
+
+### Alternatives
+
+An alternative would be to build a mutation web hook which would be able to 
+filter all object and interact with them directly. It would be more generic 
+but heavier to build. 
+This raise the question on should this be included in flux or not.
+
+## Design Details
+
+Simple update on the image automation controller should be enough. Today a 
+filter in the image policy is like: 
+
+```yaml
+    extract: $ts
+    pattern: ^pr-(?P<pr>.*)-(?P<ts>\d*)-(?P<sha1>.*)$
```

Comment:
What would happen if we'd just use regular capture groups instead of named groups? Maybe we should be able to support that as well by referencing the indices?

Reply:
I'm not found of regular capture group when the use of the capture is not in the same code space as the capture itself.   The definition is on the ImagePolicy then the use is on a comment of another kubernetes object. What's 1, 2, 3  in this context... not sure.   (I Am Not A Number, I Am A Free Man! )